src/code/z_actor.c

```diff
@@ -841,54 +841,73 @@ void Actor_Destroy(Actor* actor, PlayState* play) {
     }
 }
 
-void func_8002D7EC(Actor* actor) {
+/**
+ * Update actor's position factoring in velocity and collider displacement
+ */
+void Actor_UpdatePos(Actor* actor) {
     f32 speedRate = R_UPDATE_RATE * 0.5f;
 
     actor->world.pos.x += (actor->velocity.x * speedRate) + actor->colChkInfo.displacement.x;
     actor->world.pos.y += (actor->velocity.y * speedRate) + actor->colChkInfo.displacement.y;
     actor->world.pos.z += (actor->velocity.z * speedRate) + actor->colChkInfo.displacement.z;
 }
 
-void func_8002D868(Actor* actor) {
+/**
+ * Update actor's velocity accounting for gravity (without exceeding terminal velocity)
+ */
+void Actor_UpdateVelocityXZGravity(Actor* actor) {
     actor->velocity.x = Math_SinS(actor->world.rot.y) * actor->speedXZ;
     actor->velocity.z = Math_CosS(actor->world.rot.y) * actor->speedXZ;
 
     actor->velocity.y += actor->gravity;
+
     if (actor->velocity.y < actor->minVelocityY) {
         actor->velocity.y = actor->minVelocityY;
     }
 }
 
-void Actor_MoveForward(Actor* actor) {
-    func_8002D868(actor);
-    func_8002D7EC(actor);
+/**
+ * Move actor while accounting for its current velocity and gravity.
+ * The actor will move in the direction of its world yaw.
+ */
+void Actor_MoveXZGravity(Actor* actor) {
+    Actor_UpdateVelocityXZGravity(actor);
+    Actor_UpdatePos(actor);
 }
 
-void func_8002D908(Actor* actor) {
+/**
+ * Update actor's velocity without gravity.
+ */
+void Actor_UpdateVelocityXYZ(Actor* actor) {
     f32 sp24 = Math_CosS(actor->world.rot.x) * actor->speedXZ;
 
     actor->velocity.x = Math_SinS(actor->world.rot.y) * sp24;
     actor->velocity.y = Math_SinS(actor->world.rot.x) * actor->speedXZ;
     actor->velocity.z = Math_CosS(actor->world.rot.y) * sp24;
 }
 
-void func_8002D97C(Actor* actor) {
-    func_8002D908(actor);
-    func_8002D7EC(actor);
+/**
+ * Move actor while accounting for its current velocity.
+ * The actor will move in the direction of its world yaw and pitch.
+ */
+void Actor_MoveXYZ(Actor* actor) {
+    Actor_UpdateVelocityXYZ(actor);
+    Actor_UpdatePos(actor);
 }
 
-void func_8002D9A4(Actor* actor, f32 arg1) {
-    actor->speedXZ = Math_CosS(actor->world.rot.x) * arg1;
-    actor->velocity.y = -Math_SinS(actor->world.rot.x) * arg1;
+void Actor_SetSpeedXYZ(Actor* actor, f32 speed) {
```

Comment:
There's been a little discussion, but I think since this sets `speed` for the `XZ` variant, I would say name it `Actor_SetSpeedXZ` and than probably name the argument `speedXYZ` since that is what it takes in.

Reply:
Sounds like a good idea but what about velocity.y being set? maybe `Actor_SetSpeedXZVelY` ? <img src="https://cdn.discordapp.com/emojis/690623121664901150.webp?size=56&quality=lossless" alt="Thonk" height="24">

Reply:
my main concern with this is (all of these names) it sounding so general when it seems more specialized.
In conversations with tharo, he said that other actors probably could have made use of this function, even though its only used by projectiles in the final game. But just want to make sure that is the case before naming it so generally

Also he noted this
```
oh this in Actor_GetProjectileActor depends on projectiles using the specific function:

deltaX = Math_SinS(actor->world.rot.y) * (actor->speedXZ * 10.0f);
deltaY = actor->velocity.y + (actor->gravity * 10.0f);
deltaZ = Math_CosS(actor->world.rot.y) * (actor->speedXZ * 10.0f);
```
Im not sure this is enough to make it projectile specific, but something to note

Reply:
Hmm, to me at least the function seems kind of like a general convert a `speedXYZ` into `speedXZ` and `velocity.y` for use with the move with gravity function. I think most of the time, an actor with gravity just doesn't really care about their total speed (XYZ), but really only care about their `XZ` speed so don't end up using this function.

Reply:
(conversation continued in discord, and as of posting, still ongoing: https://discord.com/channels/688807550715560050/838852326231769089/1056077160667951154)

Reply:
This concluded as

(quoting fig)
> I’m changing the function to SetProjectileSpeed in the PR cause it’s my personal opinion of what is best, but I don’t consider this conversation resolved yet. If you guys come to a consensus on something else I’ll gladly change it

and yeah :shrug: let's go

Reply:
Yeah I dont consider it concluded, but the conversation is also not really moving anywhere. If you guys (tharo, anon, engineer, anyone else) have more to say about it then feel free